generator/mavgen_java.py

```diff
@@ -301,13 +369,24 @@ def generate_MAVLinkMessage(directory, xml_list):
     */
     public CRC crc;
 
-    public MAVLinkPacket(int payloadLength){
+    public MAVLinkPacket(int payloadLength) {
+        protocol = Protocol.${protocol_version};
+        incompat_flags = 0;
+        compat_flags = 0;
         len = payloadLength;
         payload = new MAVLinkPayload(payloadLength);
+        crc_extra = -1;
     }
 
     /**
-    * Check if the size of the Payload is equal to the "len" byte
+    * Set the MAVLink protocol for this packet.
+    */
+    public void setProtocol(Protocol protocol) {
```

Comment:
This helper isn't needed and is inconsistent with the rest of the output. All the other public fields are expected to be directly manipulated, that's why they are public.